src/goto-harness/recursive_initialization.cpp

```diff
@@ -1008,6 +1008,12 @@ code_blockt recursive_initializationt::build_function_pointer_constructor(
   for(const auto &target : targets)
   {
     auto const assign = code_assignt{dereference_exprt{result}, target};
+    auto sym_to_lookup = to_symbol_expr(to_address_of_expr(target).operands()[0]).get_identifier();
+    // skip referencing globals
```

Comment:
Please note down here that the corresponding symbols in the symbol table are no longer marked as file local, which is why you are looking them up by name I believe?